src/Features/LanguageServer/ProtocolUnitTests/CodeActions/CodeActionResolveTests.cs

```diff
@@ -420,6 +420,130 @@ class BCD
             AssertJsonEquals(expectedCodeAction, actualResolvedAction);
         }
 
+        [WpfTheory, CombinatorialData]
+        public async Task TestMoveTypeToDifferentFileInDirectory(bool mutatingLspWorkspace)
+        {
+            var markup =
+@"class ABC
+{
+}
+class {|caret:BCD|} 
+{
+}";
+
+            await using var testLspServer = await CreateTestLspServerAsync(markup, mutatingLspWorkspace, new InitializationOptions
+            {
+                ClientCapabilities = new ClientCapabilities()
+                {
+                    Workspace = new WorkspaceClientCapabilities
+                    {
+                        WorkspaceEdit = new WorkspaceEditSetting
+                        {
+                            ResourceOperations = new ResourceOperationKind[] { ResourceOperationKind.Create }
+                        }
+                    }
+                },
+
+                DocumentFileContainingFolders = new[] { Path.Combine("dir1", "dir2", "dir3") },
+            });
+
+            var unresolvedCodeAction = CodeActionsTests.CreateCodeAction(
+                title: string.Format(FeaturesResources.Move_type_to_0, "BCD.cs"),
+                kind: CodeActionKind.Refactor,
+                children: Array.Empty<LSP.VSInternalCodeAction>(),
+                data: CreateCodeActionResolveData(
+                    string.Format(FeaturesResources.Move_type_to_0, "BCD.cs"),
+                    testLspServer.GetLocations("caret").Single()),
+                priority: VSInternalPriorityLevel.Normal,
+                groupName: "Roslyn2",
+                applicableRange: new LSP.Range { Start = new Position { Line = 3, Character = 6 }, End = new Position { Line = 3, Character = 9 } },
+                diagnostics: null);
+
+            var testWorkspace = testLspServer.TestWorkspace;
+            var actualResolvedAction = await RunGetCodeActionResolveAsync(testLspServer, unresolvedCodeAction);
+
+            var existingDocument = testWorkspace.CurrentSolution.GetRequiredDocument(testWorkspace.Documents.Single().Id);
+            var existingDocumentUri = existingDocument.GetURI();
+
+            var newDocumentUri = ProtocolConversions.GetUriFromFilePath(
+                Path.Combine(Path.GetDirectoryName(existingDocument.FilePath), "BCD.cs"));
```

Comment:
Do you want to assert this also contains dir1 through dir3? My worry being this test could vacuously pass if DocumentFileContainingFolders accidentally didn't get passed through.